server.c

```
@@ -110,146 +127,173 @@ static void on_read(uv_stream_t*, ssize_t, const uv_buf_t*);
+  if (k != kh_end(mime_type)) {
+    ctype = kh_value(mime_type, k);
+  }
+  return ctype;
+}
 
-  if (response->response_offset >= response->response_size) {
-    destroy_response(response, !response->request->keep_alive);
-    return;
+static file_cache_entry*
+load_file_cache_entry(const char* path) {
+  int fd = open(path, O_RDONLY);
```

Clang (14.0.6)

[warning] variable name 'fd' is too short, expected at least 3 characters

(readability-identifier-length)